modules/new_profile/php/new_profile.class.inc

```diff
@@ -97,211 +54,83 @@ class New_Profile extends \NDB_Form
-            'dob1',
-            'Date of Birth',
-            $dateOptions,
-            $dateAttributes
-        );
-        $this->addBasicDate(
-            'dob2',
-            'Confirm Date of Birth',
-            $dateOptions,
-            $dateAttributes
-        );
-        $this->addRule(
-            array(
-             'dob1',
-             'dob2',
-            ),
-            'Date of birth fields must match',
-            'compare'
-        );
-
-        // date of birth rules
-        $this->addRule('dob1', 'Date of Birth is required', 'required');
-        $this->addRule(
-            'dob2',
-            'Date of Birth confirmation is required',
-            'required'
-        );
-
-        if ($config->getSetting('useEDC') == "true") {
-            // add expected date of confinement
-            $this->addBasicDate(
-                'edc1',
-                'Expected Date of Confinement',
-                $dateOptions,
-                $dateAttributes
-            );
-            $this->addBasicDate(
-                'edc2',
-                'Confirm EDC',
-                $dateOptions,
-                $dateAttributes
-            );
-
-            // expected date of confinement rules
-            $this->addRule(
-                array(
-                 'edc1',
-                 'edc2',
-                ),
-                'EDC fields must match',
-                'compare'
-            );
-        }
-
-        if ($config->getSetting("useProjects") == "true") {
-            $projects = \Utility::getProjectList();
-            $projList = array('' => '');
-            foreach ($projects as $projectID => $projectName) {
-                $projList[$projectID] = $projectName;
-            }
-            $this->addSelect('ProjectID', 'Project', $projList);
-        }
-
-        // add sex
-        $sexOptions = array(
-                       ''       => '',
-                       'Male'   => 'Male',
-                       'Female' => 'Female',
-                      );
-        $this->addSelect('sex', 'Sex', $sexOptions);
-
-        // sex rules
-        $this->addRule('sex', 'Sex is required', 'required');
-
-        // add sites for users with more than one site affiliation
-        $user =& \User::singleton();
-        $DB   = \Database::singleton();
+        $DB     = \Database::singleton();
+        $config = \NDB_Config::singleton();
+        $user   = \User::singleton();
+        $this->result['startYear']  = $config->getSetting('startYear');
+        $this->result['endYear']    = $config->getSetting('endYear');
+        $this->result['ageMax']     = $config->getSetting('ageMax');
+        $this->result['ageMin']     = $config->getSetting('ageMin');
+        $this->result['dobFormat']  = $config->getSetting('dobFormat');
+        $this->result['edc']        = $config->getSetting('useEDC');
+        $this->result['useProject'] = $config->getSetting('useProjects');
+        $this->result['sex']        = [
```

Comment:
apparently we like using array() instead of []

Reply:
"like" is a strong word